backend/de.metas.contracts/src/main/java/de/metas/contracts/definitive/invoicecandidate/FlatrateTermModular_DefinitiveHandler.java

```diff
@@ -0,0 +1,102 @@
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.definitive.invoicecandidate;
+
+import de.metas.contracts.FlatrateTermId;
+import de.metas.contracts.finalinvoice.invoicecandidate.FlatrateTermModular_FinalHandler;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.contracts.modular.ModCntrInvoiceType;
+import de.metas.contracts.modular.computing.ComputingResponse;
+import de.metas.contracts.modular.log.ModularContractLogDAO;
+import de.metas.contracts.modular.log.ModularContractLogQuery;
+import de.metas.contracts.modular.log.ModularContractLogService;
+import de.metas.invoicecandidate.InvoiceCandidateId;
+import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
+import de.metas.invoicecandidate.spi.IInvoiceCandidateHandler;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.compiere.SpringContextHolder;
+import org.jetbrains.annotations.NotNull;
+
+import static de.metas.contracts.modular.ModCntrInvoiceType.Definitive;
+import static de.metas.contracts.modular.ModCntrInvoiceType.Final;
+import static de.metas.invoicecandidate.spi.IInvoiceCandidateHandler.CandidatesAutoCreateMode.CREATE_CANDIDATES_AND_INVOICES;
+import static de.metas.invoicecandidate.spi.IInvoiceCandidateHandler.CandidatesAutoCreateMode.DONT;
+
+public class FlatrateTermModular_DefinitiveHandler extends FlatrateTermModular_FinalHandler
+{
+	private final ModularContractLogDAO modularContractLogDAO = SpringContextHolder.instance.getBean(ModularContractLogDAO.class);
+	private final ITrxManager trxManager = Services.get(ITrxManager.class);
+	private final ModularContractLogService modularContractLogService = SpringContextHolder.instance.getBean(ModularContractLogService.class);
+
+	@Override
+	@NonNull
+	public IInvoiceCandidateHandler.CandidatesAutoCreateMode isMissingInvoiceCandidate(@NotNull final I_C_Flatrate_Term flatrateTerm)
+	{
+		if (!flatrateTerm.isReadyForDefinitiveInvoice())
+		{
+			return DONT;
+		}
+		final boolean finalInvoiceBillableLogsExist = modularContractLogDAO.anyMatch(ModularContractLogQuery.builder()
+				.flatrateTermId(FlatrateTermId.ofRepoId(flatrateTerm.getC_Flatrate_Term_ID()))
+				.computingMethodTypes(Final.getComputingMethodTypes())
+				.processed(false)
+				.billable(true)
+				.build());
+		if (finalInvoiceBillableLogsExist)
+		{
+			return DONT;
+		}
+
+		final boolean definitiveInvoiceBillableLogsExist = modularContractLogDAO.anyMatch(ModularContractLogQuery.builder()
+				.flatrateTermId(FlatrateTermId.ofRepoId(flatrateTerm.getC_Flatrate_Term_ID()))
+				.computingMethodTypes(getModCntrInvoiceType().getComputingMethodTypes())
+				.processed(false)
+				.billable(true)
+				.build());
+
+		return definitiveInvoiceBillableLogsExist ? DONT
+				: CREATE_CANDIDATES_AND_INVOICES;
+	}
+
+	protected void afterCommit(final ComputingResponse computingResponse, final I_C_Invoice_Candidate invoiceCandidate)
```

Comment:
can't get we have a better name for "afterCommit" 
maybe it's something like after createCandidateFor....

Reply:
also pls annotate it with Override

Reply:
Changed to `processModCntrLogs`.... Now it really rolls off the tongue